rules/alerting.go

```diff
@@ -265,9 +268,11 @@ func (r *AlertingRule) forStateSample(alert *Alert, ts time.Time, v float64) pro
 	return s
 }
 
-// QueryforStateSeries returns the series for ALERTS_FOR_STATE.
-func (r *AlertingRule) QueryforStateSeries(ctx context.Context, alert *Alert, q storage.Querier) (storage.Series, error) {
-	smpl := r.forStateSample(alert, time.Now(), 0)
+// QueryforStateSeries returns the series for ALERTS_FOR_STATE of the alert rule.
+func (r *AlertingRule) QueryforStateSeries(ctx context.Context, q storage.Querier) (storage.SeriesSet, error) {
```

Comment:
[nit] Given we're changing the signature anyway, any reason why the "for" was lower case instead of being `QueryForStateSeries()`?

Reply:
Done in https://github.com/prometheus/prometheus/pull/13980/commits/ccfafae36d4a0d49a4b099fafbe56eb66ffcb6bb